packages/components/src/hooks/useSnackbar.tsx

```diff
@@ -0,0 +1,45 @@
+import { useContext } from 'react'
+
+import * as Toast from 'react-native-toast-notifications'
+
+import {
+  SNACKBAR_DEFAULT_OFFSET,
+  modifyToastOptions,
+} from '../components/Snackbar/Snackbar'
+import { SnackbarContext } from '../components/Snackbar/SnackbarProvider'
+
+/**
+ * Passthrough of useToast hook which has been modified to allow for optional
+ * offset setting
+ */
+export function useSnackbar() {
+  const toast = Toast.useToast()
+  const context = useContext(SnackbarContext)
+
+  if (!context) {
+    throw new Error('useSnackbar must be used within a SnackbarProvider')
+  }
+
+  const { offset, setOffset } = context
+
+  const show = (
+    message: string,
+    snackbarOptions?: modifyToastOptions['data'],
```

Comment:
Hmmm. With the revisions I wonder if we should change this up now. 

Previously, we were leaning more directly on the library and correspondingly couldn't modify the structure as well. With how this is now, if we wanted to allow modifying the duration in the future we could easily bundle it into the data and would just update the `toast.show` call below to send it in alongside data so it'd be a non-breaking change. 

So, unless I'm mistaken I think we could do some restructuring to simplify a bit/make it more intuitive. Thoughts on splitting out a ticket to rework this a bit? A bit tangential to this ticket.

Reply:
Created ticket #439